### PR TITLE
fix: target management clusters in SSS

### PR DESCRIPTION
**Pull request by jan--f**
Signed-off-by: Jan Fajerski <jfajersk@redhat.com>

### CODE REVIEW

**Comment by apahim** [comment]
MON-2677 is about deploying OBO to the entire OSD/ROSA fleet, which can be achieved by using the label:

```
api.openshift.com/managed: "true"
```

Why are we targeting Management Clusters only?

**Comment by jan--f** [comment]
> MON-2677 is about deploying OBO to the entire OSD/ROSA fleet, which can be achieved by using the label:
> 
> ```
> api.openshift.com/managed: "true"
> ```
> 
> Why are we targeting Management Clusters only?

My bad, fixed now

File: deploy/olm/syncselector-template.yaml [hunk 1]
```diff
@@ -22,7 +22,8 @@ objects:
     spec:
       clusterDeploymentSelector:
         matchLabels:
-          api.openshift.com/addon-reference-addon: 'true'
+          ext-hypershift.openshift.io/cluster-type: management-cluster
+          ext-hypershift.openshift.io/cluster-type: service-cluster
```
Comment (apahim) [inline deploy/olm/syncselector-template.yaml:26]: We should use just:
```
api.openshift.com/managed: "true"
```
Nothing else, so it hits the entire fleet.

Comment (maorfr) [reply deploy/olm/syncselector-template.yaml:26]: https://issues.redhat.com/browse/SDA-6646?focusedCommentId=20924957&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-20924957

Comment (maorfr) [reply deploy/olm/syncselector-template.yaml:26]: After reading this comment, +1 to what @apahim is suggesting

Comment (wanghaoran1988) [reply deploy/olm/syncselector-template.yaml:26]: My understanding is the two labels listed here is a "AND" operation, there is no cluster will has these two labels together.

Comment (maorfr) [reply deploy/olm/syncselector-template.yaml:26]: we've concluded over slack that @apahim's proposal is the way to go.

**Review by maorfr — APPROVED** [review]
/lgtm

**Review by sthaha — APPROVED** [review]

**Pull request merged**